src/include/txn_inline.h

```diff
@@ -909,6 +910,16 @@ __wt_upd_alloc(WT_SESSION_IMPL *session, const WT_ITEM *value, u_int modify_type
         (value != NULL &&
           !(modify_type == WT_UPDATE_RESERVE || modify_type == WT_UPDATE_TOMBSTONE)));
 
+    /*
+     * Ensure that the memory allocated is never less than the size reported by sizeof. Otherwise
+     * bit-exact tools like Google MSan may infer the structure is not completely initialized.
+     */
+    if (value != NULL) {
+        allocsz = WT_MAX(WT_UPDATE_SIZE + value->size, sizeof(*upd));
+    } else {
+        allocsz = sizeof(*upd);
```

Comment:
Hi Hari,
I have different information. In the current version the structure is 47 bytes size.
```
#define WT_UPDATE_SIZE 47
```
The compiler adds trailing padding to ensure 8 byte alignment.
```
(gdb) p sizeof(struct __wt_update)
$1 = 48
```
Given that MSan states it is [bit-exact](https://github.com/google/sanitizers/wiki/MemorySanitizer): I assume it is using symbol generated by the compiler `-fsanitize=memory`.

Reply:
Yes, `WT_UPDATE_SIZE` doesn't include the last byte that holds the data. 
I am not against the use of `sizeof(WT_UPDATE)` structure as it allocates an extra one byte for every tombstone that we allocate.

My concern is that in future if we change the WT_UPDATE structure and it may lead to allocating more unused bytes due to the `sizeof` usage. Currently, whenever an update is allocated, we do perform size alignment. The extra bytes allocation shouldn't be a problem, I feel better to check it whenever we change the structure size. 

How about adding a comment to validate the size in `__wt_upd_alloc` whenever that structure size changes in future?

Reply:
The intention is to allocate as many bytes as needed to fulfil any trailing padding in the `struct`, so in the range [0, 7] bytes. This is to ensure the allocation matches `sizeof()`.

I'm only peripherally aware of tombstones, and unsure as to what percentage of update structs allocated they make up. Are you concerned about space usage?

If by size alignment you are referring to `WT_UPDATE_MEMSIZE` that currently rounds up to the nearest size in bytes divisible by 32. Where as the existing size check in `src/include/verify_build.h` rounds up to the nearest size in bytes divisible by 8.

There is already a check in `src/include/verify_build.h` so if/when the size changes.

Reply:
Thanks for the details.